src/utilcode/loaderheap.cpp

```diff
@@ -727,18 +728,42 @@ struct LoaderHeapFreeBlock
 #endif
 
             INDEBUG(memset(pMem, 0xcc, dwTotalSize);)
+
             LoaderHeapFreeBlock *pNewBlock = (LoaderHeapFreeBlock*)pMem;
-            pNewBlock->m_pNext  = *ppHead;
             pNewBlock->m_dwSize = dwTotalSize;
-            *ppHead = pNewBlock;
 
-            MergeBlock(pNewBlock, pHeap);
+            if (pHeap->IsFIFO())
```

Comment:
I am wondering how this would can interact with other places that use the executable heap, and whether we can make this generally more reliable.

Would it be better to move the FIFO into one level up to where UMEntryThunks are allocate? I am thinking about something like:
```
UMEntryThunk::CreateUMEntryThunk
{
    if (number of cached thunks < 100)
        Allocate a new thunk using GetGlobalLoaderAllocator()->GetExecutableHeap()
    else
        Use thunk from the LIFO cache
}

UMEntryThunk::Terminate
{
    Add a thunk to LIFO cache
}
```